pkg/pillar/cmd/zedmanager/handleresolve.go

```diff
@@ -0,0 +1,153 @@
+	pub.Publish(key, *config)
+	log.Debugf("publishResolveConfig(%s) Done\n", key)
+}
+
+func unpublishResolveConfig(ctx *zedmanagerContext,
+	config *types.ResolveConfig) {
+
+	key := config.Key()
+	log.Debugf("unpublishResolveConfig(%s)\n", key)
+	pub := ctx.pubAppImgResolveConfig
+	pub.Unpublish(key)
+	log.Debugf("unpublishResolveConfig(%s) Done\n", key)
+}
+
+func lookupResolveConfig(ctx *zedmanagerContext,
+	key string) *types.ResolveConfig {
+
+	log.Infof("lookupResolveConfig(%s)\n", key)
+	pub := ctx.pubAppImgResolveConfig
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveConfig(%s) not found\n", key)
+		return nil
+	}
+	config := c.(types.ResolveConfig)
+	log.Infof("lookupResolveConfig(%s) Done\n", key)
+	return &config
+}
+
+func lookupResolveStatus(ctx *zedmanagerContext,
+	key string) *types.ResolveStatus {
+
+	log.Infof("lookupResolveStatus(%s)\n", key)
+	sub := ctx.subAppImgResolveStatus
+	c, _ := sub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveStatus(%s) not found\n", key)
+		return nil
+	}
+	status := c.(types.ResolveStatus)
+	log.Infof("lookupResolveStatus(%s) Done\n", key)
+	return &status
+}
+
+func deleteResolveConfig(ctx *zedmanagerContext, key string) {
+	log.Infof("deleteResolveConfig for %s\n", key)
+	rc := lookupResolveConfig(ctx, key)
+	if rc != nil {
+		log.Infof("deleteResolveConfig for %s found\n", key)
+		unpublishResolveConfig(ctx, rc)
+	}
+	log.Infof("deleteResolveConfig for %s Done\n", key)
+}
+
+func handleResolveStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleResolveStatusModify for %s\n", key)
+	ctx := ctxArg.(*zedmanagerContext)
+	rs := statusArg.(types.ResolveStatus)
+	pub := ctx.pubAppInstanceStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		status := st.(types.AppInstanceStatus)
+		config := lookupAppInstanceConfig(ctx, status.Key())
+		if config == nil {
+			errStr := fmt.Sprintf("App Instance config not found for %s while resolving tags\n",
+				status.Key())
+			log.Error(errStr)
+			status.SetError(errStr, time.Now())
+			publishAppInstanceStatus(ctx, &status)
+			continue
+		}
+		for i := range status.StorageStatusList {
+			ss := &status.StorageStatusList[i]
+			if !ss.HasResolverRef || ss.Name != rs.Name || ss.DatastoreID != rs.DatastoreID {
+				continue
+			}
+			log.Infof("Processing ResolveStatus for storage status (%v) in app instance (%v)\n",
+				ss.ImageID, status.UUIDandVersion.UUID)
+			if len(rs.Error) != 0 {
+				errStr := fmt.Sprintf("Error occurred while resolving tags, updating app instance %s: %v\n",
+					status.Key(), rs.Error)
+				log.Error(errStr)
+				status.SetErrorWithSource(rs.Error, types.ResolveStatus{}, rs.ErrorTime)
+				publishAppInstanceStatus(ctx, &status)
+				continue
+			}
+			if rs.ImageSha256 == "" {
+				errStr := fmt.Sprintf("Empty SHA in resolve status, updating app instance %s.\n",
+					status.Key())
+				log.Error(errStr)
+				status.SetError(errStr, time.Now())
```

Comment:
If you want to be able to clear this error should the Resolver later send a ResolveStatus with a sha, you need to mark with using SetErrorWithSource as above.

Reply:
Resolved